src/test/cool/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+
+project(test_cool LANGUAGES CSharp)
+
+# CMAKE doesn't have a good way to deal with tools which may
+# generate multiple files not known a priori. See
+# https://stackoverflow.com/questions/44076307/cmake-globbing-generated-files
+# We could try to extract this list from "ildasm /text foo.winmd".
+# For now, let's hardcode.
+set(generated_files
+    ${CMAKE_CURRENT_BINARY_DIR}/base-interop.cs
+    ${CMAKE_CURRENT_BINARY_DIR}/TestCool.BasicEnum.cs
+    ${CMAKE_CURRENT_BINARY_DIR}/TestCool.BasicStruct.cs
+    ${CMAKE_CURRENT_BINARY_DIR}/TestCool.Class.cs
+    ${CMAKE_CURRENT_BINARY_DIR}/TestCool.IClass.cs
+    ${CMAKE_CURRENT_BINARY_DIR}/TestCool.IInterface.cs)
+
+add_executable(test_cool "")
+target_sources(test_cool
+    PRIVATE main.cs
+    testcool.idl
+    ${generated_files}
+)
+set(CMAKE_CSharp_FLAGS "/langversion:7.1") # otherwise cmake defaults to 3
+set_target_properties(test_cool PROPERTIES VS_DOTNET_TARGET_FRAMEWORK_VERSION "4.5.1")
+set_target_properties(test_cool PROPERTIES VS_GLOBAL_LangVersion "7.1")
+set_target_properties(test_cool PROPERTIES VS_GLOBAL_AllowUnsafeBlocks "true")
+
+
+# Inputs and outputs. MIDL cannot handle / in paths.
+file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR}\\testcool.idl   TestCool_idl)
+file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR}\\testcool.winmd TestCool_winmd)
+
+if( NOT EXISTS "$ENV{VS140COMNTOOLS}")
+    message(FATAL_ERROR "VS140COMNTOOLS environment variable is not defined")
+endif()
+
+# Find WindowsKitsRoot10 (from registry)
+get_filename_component(WindowsKitsRoot10 "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots;KitsRoot10]" ABSOLUTE)
+if(NOT EXISTS "${WindowsKitsRoot10}")
+    message(FATAL_ERROR "Could not find the Windows SDK Root")
+endif()
+
+# Find WindowsSDKContractDir (from CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION)
+file(TO_NATIVE_PATH
+    ${WindowsKitsRoot10}/References/${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}/Windows.Foundation.FoundationContract/3.0.0.0
```

Comment:
this could be hard to maintain - why not just use the VS generator and allow it to automatically find the SDK, etc?